src/processor/snapshot/database.rs

```diff
@@ -0,0 +1,139 @@
+use std::{convert::TryFrom, path::PathBuf};
+
+use ethers::types::H256;
+use eyre::Result;
+use rocksdb::{Options, DB};
+use thiserror::Error;
+
+use super::types::{SnapshotFactoryDependency, SnapshotStorageLog};
+
+const STORAGE_LOGS: &str = "storage_logs";
+const INDEX_TO_KEY_MAP: &str = "index_to_key_map";
+const FACTORY_DEPS: &str = "factory_deps";
+const METADATA: &str = "metadata";
+
+const LAST_REPEATED_KEY_INDEX: &str = "LAST_REPEATED_KEY_INDEX";
+
+#[allow(clippy::enum_variant_names)]
+#[derive(Error, Debug)]
+pub enum DatabaseError {
+    #[error("no such key")]
+    NoSuchKey,
+}
+
+pub struct SnapshotDB {
+    db: DB,
+}
+
+impl SnapshotDB {
+    pub fn new(db_path: PathBuf) -> Result<Self> {
+        let mut db_opts = Options::default();
+        db_opts.create_missing_column_families(true);
+        db_opts.create_if_missing(true);
+
+        let db = DB::open_cf(
+            &db_opts,
+            db_path,
+            vec![METADATA, STORAGE_LOGS, INDEX_TO_KEY_MAP, FACTORY_DEPS],
+        )?;
+
+        Ok(Self { db })
+    }
+
+    pub fn get_last_repeated_key_index(&self) -> Result<u64> {
+        let metadata = self.db.cf_handle(METADATA).unwrap();
```

Comment:
```suggestion
        let metadata = self.db.cf_handle(METADATA)?;
```

Reply:
There are quite a few `unwrap`s even in functions that return a `Result`, any particular reason for this?

Reply:
No particular reason. Initially I had a bit of a hassle with the rocksdb as I accidentally followed API docs from two different crates which were close and then I was also iterating on necessary API here quite a bit. :grimacing:

Reply:
Looking this specific case again after my compiler barked at me :smile: 

`db.cf_handle()` returns `Option`, but I know that the handle is there, because it's inserted there in the constructor. This can never fail, unless if the DB is fucked underneath, in which case the program should panic anyway.